### PR TITLE
storage: add/fix revision/generation description

### PR DESCRIPTION
**Pull request by gyuho**
This adds some struct descriptions to revision/generation structs and corrects
some spelling errors.

### CODE REVIEW

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
Thanks. Will merge after test pass.

File: storage/revision.go [hunk 1]
```diff
@@ -21,9 +21,15 @@ import "encoding/binary"
 // is a '_'. The last 8 bytes is the revision.sub in big-endian format.
 const revBytesLen = 8 + 1 + 8
 
+// revision is an index for key/value operations,
```
Comment (xiang90) [inline storage/revision.go]: A revision indicates a modify of the key-value space.
The set of changes that share same main revision changed the key-value space atomically.

File: storage/key_index.go [hunk 2]
```diff
@@ -294,6 +294,8 @@ func (ki *keyIndex) String() string {
 	return s
 }
 
+// generation contains multiple versions of revision
```
Comment (xiang90) [inline storage/key_index.go]: one line please

File: storage/revision.go [hunk 3]
```diff
@@ -21,9 +21,15 @@ import "encoding/binary"
...
+// useful for tracking versions of keys.
 type revision struct {
+	// main revision increases whenever request is applied.
```
Comment (xiang90) [inline storage/revision.go]: this is not accurate. main is the main revision of a set of changes that happens atomically.

File: storage/revision.go [hunk 4]
```diff
@@ -21,9 +21,15 @@ import "encoding/binary"
...
 	main int64
-	sub  int64
+
+	// sub is an incremental index within one transaction with the same main
```
Comment (xiang90) [inline storage/revision.go]: sub is the the sub revision of a change in a set of changes that happens atomically. Each changes have different increasing sub revision in that set.

Comment (gyuho) [inline storage/revision.go]: Will fix. Thanks,

**Comment by gyuho** [comment]
@xiang90 Just fixed them. PTAL. Thanks,

File: storage/key_index.go [hunk 5]
```diff
@@ -294,6 +294,7 @@ func (ki *keyIndex) String() string {
 	return s
 }
 
+// generation contains multiple versions of revision of one key.
```
Comment (xiang90) [inline storage/key_index.go]: generation contains multiple revisions of a key.

**Comment by gyuho** [comment]
@xiang90 Just fixed. PTAL. Thanks!

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
Thanks, will merge after test pass.

**Pull request merged**